src/nmodl/parsact.cpp

```diff
@@ -76,28 +76,23 @@ void explicit_decl(int level, Item* q) {
     Symbol* sym;
 
     sym = SYM(q);
-    if (!level) { /* No multiple declarations at the root level and
-            the symbol is marked explicitly declared */
-        if (sym->usage & EXPLICIT_DECL) {
-            diag("Multiple declaration of ", sym->name);
-        }
-        sym->usage |= EXPLICIT_DECL;
+    if (sym->usage & EXPLICIT_DECL) {
+        diag("Multiple declaration of ", sym->name);
     }
+    sym->usage |= EXPLICIT_DECL;
     /* this ensures that declared PRIMES will appear in .var file */
     sym->usage |= DEP;
 
-    if (level >= sym->level) {
+    if (sym->level) {
         assert(previous_str);
     }
     /* resolve possible type conflicts */
-    if (type_change(sym, level)) {
+    if (type_change(sym)) {
         return;
     }
     /* resolve which declaration takes precedence */
-    if (level < sym->level) { /* new one takes precedence */
-        sym->level = level;
-    } else if (level > sym->level) { /* old one takes precedence */
-        sym->u.str = previous_str;
+    if (!sym->level) { /* new one takes precedence */
+        sym->level = 1;
     } else if (strcmp(sym->u.str, previous_str) != 0) { /* not identical */
```

Comment:
I guess ```previous_str``` needs a specific review. Here, since the level can be the same, it seems to refer to duplicate definitions.

Reply:
eliminated